File: dkplayer-players/exo/src/main/java/xyz/doikki/videoplayer/exo/ExoMediaPlayerFactory.java
package xyz.doikki.videoplayer.exo;

import android.content.Context;

import xyz.doikki.videoplayer.player.PlayerFactory;

public class ExoMediaPlayerFactory implements PlayerFactory {

    public static ExoMediaPlayerFactory create() {
        return new ExoMediaPlayerFactory();
    }

    @Override
    public ExoMediaPlayer create(Context context) {
        return new ExoMediaPlayer(context);
    }
}

File: dkplayer-players/exo/src/main/java/xyz/doikki/videoplayer/exo/ExoMediaPlayerFactory.kt
package xyz.doikki.videoplayer.exo

import android.content.Context
import xyz.doikki.videoplayer.player.PlayerFactory

class ExoMediaPlayerFactory : PlayerFactory {
    override fun create(context: Context): ExoMediaPlayer {
        return ExoMediaPlayer(context)
    }

    companion object {
        @JvmStatic
        fun create(): ExoMediaPlayerFactory {
            return ExoMediaPlayerFactory()
        }
    }
}